pluto-message-ingestion/pluto-message-processor.js

```diff
@@ -0,0 +1,95 @@
+class PlutoMessageProcessor {
+  constructor({ hostname, hmacRequest, logger }) {
+    this.hostname = hostname;
+    this.hmacRequest = hmacRequest;
+    this.logger = logger;
+  }
+
+  process(message) {
+    switch (message.type) {
+      case 'project-created': {
+        const project = PlutoMessageProcessor._convertMessageToProject(message);
+
+        return new Promise((resolve, reject) => {
+          this._createProject(project)
+            .then(resp => {
+              this.logger.log('successfully created project', {
+                response: resp
+              });
+              resolve(resp);
+            })
+            .catch(err => {
+              this.logger.error('failed to create project', {
+                status: err.status,
+                response: err.response,
+                project: project
+              });
+              reject(err);
+            });
+        });
+      }
+      case 'project-updated': {
+        const project = PlutoMessageProcessor._convertMessageToProject(message);
+
+        return new Promise((resolve, reject) => {
+          this._updateProject(project)
+            .then(resp => {
+              this.logger.log('successfully updated project', {
+                response: resp
+              });
+              resolve(resp);
+            })
+            .catch(err => {
+              const logDetail = {
+                status: err.status,
+                response: err.response,
+                project: project
+              };
+
+              if (err.status === 404) {
```

Comment:
A little painful...

This will occur when a pre-existing project in Pluto gets updated. Obviously, MAM doesn't know of pre-existing projects, but will still receive a message. Discarding the message effectively means MAM can only Direct Upload to projects created **after** these changes go out.

Need to chat with @fredex42 and Editorial to confirm the lifespan of a project - if a project can span a few months we maybe want to relax this restraint entirely and ingest all messages into Dynamo otherwise Editorial would have to create a new Pluto project just to Direct Upload.